src/hooks/useFetch.js

```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+const useFetch = (url) => {
+  const [data, setData] = React.useState([]);
+  React.useEffect(() => {
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => setData(data));
+  }, [url]);
+
+  return [data, setData];
```

Comment:
no need to return the setData, we don't what to give user the ability to update the api data here